src/evaluators/UnaryExpression.js

```diff
@@ -69,8 +69,13 @@ function evaluateDeleteOperation(expr: Value | Reference, realm: Realm) {
     // b. Let baseObj be ! ToObject(GetBase(ref)).
     let base = Environment.GetBase(realm, ref);
     if (base instanceof AbstractValue && !(base instanceof AbstractObjectValue)) {
-      // if the base is abstract but not an abstract object, we have a type error
-      throw realm.createErrorThrowCompletion(realm.intrinsics.TypeError);
+      let error = new CompilerDiagnostic(
+        "might not be an object, hence this unary operation might throw a runtime TypeError",
+        ast.argument.loc,
+        "PP0027",
```

Comment:
PP0012 seems more to the point here.

Reply:
I removed that FatalError altogether. We can follow up the delete logic in another PR.